crates/edge-registry/src/args.rs

```diff
@@ -0,0 +1,101 @@
+use std::str::FromStr;
+use structopt::StructOpt;
+use utils::communication::consumer::CommonConsumerConfig;
+use utils::metrics;
+
+#[derive(Clone, Debug, StructOpt)]
+pub struct RegistryConfig {
+    #[structopt(long, env)]
+    pub postgres_username: String,
+    #[structopt(long, env)]
+    pub postgres_password: String,
+    #[structopt(long, env)]
+    pub postgres_host: String,
+    #[structopt(long, env, default_value = "5432")]
+    pub postgres_port: u16,
+    #[structopt(long, env)]
+    pub postgres_dbname: String,
+    #[structopt(long, env, default_value = "public")]
+    pub postgres_schema: String,
+    #[structopt(long, env, default_value = "50110")]
+    /// gRPC server port to host edge-registry on
+    pub communication_port: u16,
+    #[structopt(long, env, default_value = metrics::DEFAULT_PORT)]
+    /// Prometheus metrics port
+    pub metrics_port: u16,
+    #[structopt(flatten)]
+    pub consumer_config: ConsumerConfig,
+}
+
+#[derive(Clone, Debug, StructOpt)]
+#[cfg_attr(test, derive(PartialEq))]
+pub enum ConsumerMethod {
+    Kafka,
+    Amqp,
+}
+
+#[derive(Clone, Debug, StructOpt)]
+pub struct ConsumerConfig {
+    #[structopt(long, env)]
```

Comment:
Id recommend
```suggestion
    #[structopt(long, env = "CONSUMER_METHID")]
```
or
```suggestion
    #[structopt(long, env = "MQ_METHOD")]
```
because generic `METHOD` env variable is... well to generic